Dockerfile

```diff
@@ -0,0 +1,10 @@
+FROM node:14-alpine
+ENV NODE_ENV=development
+WORKDIR /usr/src/app
+COPY ["package.json", "yarn.lock", "./"]
+RUN yarn install
+COPY . .
```

Comment:
Why are there two COPY commands?

Reply:
It's 2 stages, for docker layer caching reasons. If the package json and yarn lock haven't changed, docker doesn't have to rerun those stages, and if the rest of the copied stuff hasn't changed either then that can skip too. Ideally it would be even more granular than copy . . And only copy the stuff needed for building

Reply:
Nice! Thanks for the background, I'm still getting familiar with Node & the JS ecosystem.